# Part -1 Python Basics (Variables)

1. Print Your Name with your Father name and Date of birth using suitable escape sequence charactor

In [1]:
name='Umar Khan'
fathername='Muhammed Nazeer Khan Ghouri'
DOB='08/03/2005'
print(f"My name is {name}\nMy Father Name is {fathername}\nMy Date of Birth is {DOB}")

My name is Umar Khan
My Father Name is Muhammed Nazeer Khan Ghouri
My Date of Birth is 08/03/2005


2. Write your small bio using variables and print it using print function

In [2]:
name="Umar khan Ghouri"
educationastatus="BSE First Year"
interest="Interested In Learning Coding"
hobbies='Learning to code'
print(f"Name: {name}\nHobby: {hobbies}\ninterest: {interest}\nEducational status: {educationastatus}")

Name: Umar khan Ghouri
Hobby: Learning to code
interest: Interested In Learning Coding
Educational status: BSE First Year


3. Write a program in which use all the operators we can use in Python

In [50]:
# Arithmetic Operators
a = 50
b = 100
print(f"Addition: {a} + {b} = {a + b}")
print(f"Subtraction: {a} - {b} = {a - b}")
print(f"Multiplication: {a} * {b} = {a * b}")
print(f"Division: {a} / {b} = {a / b}")
print(f"Floor Division: {a} // {b} = {a // b}")
print(f"Modulus: {a} % {b} = {a % b}")
print(f"Exponentiation: {a} ** {b} = {a ** b}")

# Comparison Operators
print(f"Equal: {a} == {b} = {a == b}")
print(f"Not Equal: {a} != {b} = {a != b}")
print(f"Greater Than: {a} > {b} = {a > b}")
print(f"Less Than: {a} < {b} = {a < b}")
print(f"Greater Than or Equal To: {a} >= {b} = {a >= b}")
print(f"Less Than or Equal To: {a} <= {b} = {a <= b}")

# Logical Operators
x = True
y = False
print(f"Logical AND: {x} and {y} = {x and y}")
print(f"Logical OR: {x} or {y} = {x or y}")
print(f"Logical NOT: not {x} = {not x}")

# Assignment Operators
c = a + b
print(f"Assignment: c = {c}")
c += a
print(f"Add and Assign: c += {a} = {c}")
c -= b
print(f"Subtract and Assign: c -= {b} = {c}")
c *= a
print(f"Multiply and Assign: c *= {a} = {c}")
c /= b
print(f"Divide and Assign: c /= {b} = {c}")
c %= a
print(f"Modulus and Assign: c %= {a} = {c}")
c //= b
print(f"Floor Divide and Assign: c //= {b} = {c}")

# Membership Operators
my_list = [10, 22, 33, 44, 55]
print(f"(in): 33 in list = {33 in my_list}")
print(f"(not in): 66 not in list = {66 not in my_list}")

# Identity Operators
d = a
print(f"Identity (is): a is d = {a is d}")
print(f"Identity (is not): a is not b = {a is not b}")


Addition: 50 + 100 = 150
Subtraction: 50 - 100 = -50
Multiplication: 50 * 100 = 5000
Division: 50 / 100 = 0.5
Floor Division: 50 // 100 = 0
Modulus: 50 % 100 = 50
Exponentiation: 50 ** 100 = 78886090522101180541172856528278622967320643510902300477027893066406250000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
Equal: 50 == 100 = False
Not Equal: 50 != 100 = True
Greater Than: 50 > 100 = False
Less Than: 50 < 100 = True
Greater Than or Equal To: 50 >= 100 = False
Less Than or Equal To: 50 <= 100 = True
Logical AND: True and False = False
Logical OR: True or False = True
Logical NOT: not True = False
Assignment: c = 150
Add and Assign: c += 50 = 200
Subtract and Assign: c -= 100 = 100
Multiply and Assign: c *= 50 = 5000
Divide and Assign: c /= 100 = 50.0
Modulus and Assign: c %= 50 = 0.0
Floor Divide and Assign: c //= 100 = 0.0
(in): 33 in list = True
(not in): 66 not in list = True
Identity (is): a is d = True
Identity (is not): a is not

4. Completes the following steps of small task:
    - Mention Marks of English , Islamiat and Maths out of 100 in 3 different variables
    - Mention Variable of Total Marks and assign 300 to it
    - Calculate Percentage

In [3]:
EnglishMarks=65
IslamiatMarks=77
MathsMarks=99
TotalMarks=300
sum=EnglishMarks+IslamiatMarks+MathsMarks
percentage=(sum/TotalMarks)*100
print(percentage)

80.33333333333333


# Part -2 Python Basics (Conditional Statements)

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [4]:
usersalary=int(input("Enter Your Salary:"))
userYear=int(input("Enter Your Year Of Service:"))
bonususersalary=usersalary*0.05
newsalary=usersalary+bonususersalary
if userYear>5:
    print("Congrats 5% Bonus will be Added to Your Salary")
    print(f"The New Amount is:${newsalary}")
else:
    print("Sorry Your Year of Service is Not Greater Than 5 year So you are not eligible")
    
    

Enter Your Salary: 5000
Enter Your Year Of Service: 6


Congrats 5% Bonus will be Added to Your Salary
The New Amount is:$5250.0



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [5]:
userage=int(input("Enter Your Age"))
if userage>17:
    print("You are Eligible For Voting")
else:
    print("You are not Eligible For Voting")
    

Enter Your Age 17


You are not Eligible For Voting



3) Write a program to check whether a number entered by user is even or odd.

In [6]:
usernumber=int(input('Enter Your Number'))
if usernumber%2==0:
    print(f'Number {usernumber} is Even')
else:
    print(f"Number {usernumber} is ODD")

Enter Your Number 6


Number 6 is Even


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [7]:
usernumber=int(input('Enter Your Number'))
if usernumber%7==0:
    print(f'Number {usernumber} is Divisible by 7')
else:
    print(f"Number {usernumber} is Not Divisible By 7")

Enter Your Number 99


Number 99 is Not Divisible By 7


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [8]:
usernumber=int(input('Enter Your Number'))
if usernumber%5==0:
    print(f'Hello')
else:
    print(f"Bye")
    

Enter Your Number 65


Hello


7) Write a program to display the last digit of a number.


In [9]:
usernumber=int(input('Enter Your Number'))
lastdigit=usernumber%10
print(f'The Last digit of Your Number is {lastdigit}')


Enter Your Number 69


The Last digit of Your Number is 9


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [10]:
userlenght=int(input("Enter the Value of Lenght"))
userbreath=int(input("Enter the Value of Breath"))
if userbreath==userlenght:
    print("This is A Square")
else:
    print("This is A Rectangle")

Enter the Value of Lenght 25
Enter the Value of Breath 36


This is A Rectangle


10) Take two int values from user and print greatest among them.

In [11]:
uservalue1=int(input("Enter the First Value"))
uservalue2=int(input("Enter the Second Value"))
if uservalue1>uservalue2:
    print(f"This value which is {uservalue1} is greater than {uservalue2}")
else:
    print(f"This value which is {uservalue2} is greater than {uservalue1} ")

Enter the First Value 56
Enter the Second Value 59


This value which is 59 is greater than 56 


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [15]:
userQuantity=int(input("Enter Your Quantity:"))
price1=100
newprice=userQuantity*price1
print(f"Your amount is ${newprice}")
discount=newprice*0.10
discountedprice=newprice-discount
if userQuantity>1000:
    print("Since you purchsed more than 1000 in Quantity Congrats 10% Discount will be Added")
    print(f"The New Amount is:${discountedprice}")
else:
    print("Sorry Your Purchased Quantity is Not Greater Than 1000 So You will Not Get a Disount")
    
    

Enter Your Quantity: 1000


Your amount is 100000
Sorry Your Purchased Quantity is Not Greater Than 1000 So You will Not Get a Disount


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [16]:
usergrade=(int(input("Enter Your Grade")))
if usergrade<=25:
    print("Your Grade is F ,You Need To Work Hard")
elif usergrade<=45:
    print("Your Grade is E ,You Need To Work Hard")
elif usergrade<=50:
    print("Your Grade is D ,You Need To Work Hard")
elif usergrade<=60:
    print("Your Grade is C ,Need Improvement")
elif usergrade<=80:
    print("Your Grade is B ,Good Work")
elif usergrade>80:
    print("Your Grade is A ,Execellent")

Enter Your Grade 56


Your Grade is C ,Need Improvement


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

- Number of classes held

- Number of classes attended.

And print

- percentage of class attended

- Is student is allowed to sit in exam or not.


In [17]:
number_of_clases_held=int(input('Enter the Number of Clases Held'))
number_of_clases_attended=(int(input('Enter the number of Clases You have Attended')))
attended=(number_of_clases_attended/number_of_clases_held)*100
if attended<75:
    print('You are not Eligible For taking This Exam')
else:
    print('You are Eligible For taking This Exam')

Enter the Number of Clases Held 12
Enter the number of Clases You have Attended 8


You are not Eligible For taking This Exam


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [18]:
number_of_clases_held=int(input('Enter the Number of Clases Held'))
number_of_clases_attended=(int(input('Enter the number of Clases You have Attended')))
attended=(number_of_clases_attended/number_of_clases_held)*100
medical_emergency=(input('Did you Had Any Medical Emergency, Enter Y/N')).strip().lower()
if medical_emergency=='y':
    print('Since You Had An Medical Emergency You can Take This Test')
elif attended<75:
    print('You are not Eligible For taking This Exam')
else:
    print('You are Eligible For taking This Exam')

Enter the Number of Clases Held 12
Enter the number of Clases You have Attended 8
Did you Had Any Medical Emergency, Enter Y/N y


Since You Had An Medical Emergency You can Take This Test


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [20]:
userLeapYear=int(input('Enter Your YEAR'))
if userLeapYear%4==0 and userLeapYear%400==0:
    print(f'The Year is a leap Year {userLeapYear}')
else:
    print("This is Not A leap Year")

Enter Your YEAR 2000


The Year is a leap Year 2000


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [25]:
age=int(input("Enter Your Age"))
gender=input("Enter Your Gender M/F").strip().lower()
marital_status=input("Enter Your marital status Y/N").strip().lower()
if gender=="f":
    print('You can only work in Urban Areas')
elif gender=='m' and 20<=age<40:
    print('You are eligible to work anywhere')
elif gender=='m' and 40<=age<60:
    print ('You are eligible for work only in Urban Areas')
else:
    print("Error")

Enter Your Age 20
Enter Your Gender M/F m
Enter Your marital status Y/N y


You are eligible to work anywhere


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

In [30]:
number_of_unit=int(input("Enter Your Units"))
if number_of_unit<100:
    print('This Month You are charged With 0$')
elif 100<=number_of_unit<200:
    number_of_unit_above_100=number_of_unit*5
    print(f'This Month You are charged With {number_of_unit_above_100}$')
elif number_of_unit>=200:
    number_of_unit_above_200=number_of_unit*10
    print(f'This Month You are charged With {number_of_unit_above_200}$')
else:
    print('Enter a Valid Amount')



Enter Your Units 350


This Month You are charged With 3500$


13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [46]:
first_user_age=int(input('Enter The Age of First User'))
second_user_age=int(input('Enter The Age of Second User'))
third_user_age=int(input('Enter The Age of Third User'))
if first_user_age >100 and second_user_age>100 and third_user_age>100:
    print('This is Not A Valid Age')
else:
    if first_user_age>second_user_age and first_user_age> third_user_age:
        print(F"{first_user_age} is the Oldest")
    elif second_user_age> first_user_age and second_user_age> third_user_age:
        print(F"{second_user_age} is the Oldest")
    elif third_user_age>second_user_age and third_user_age>first_user_age:
        print(F"{third_user_age} is the Oldest")
    if first_user_age<second_user_age and first_user_age<third_user_age:
        print(F"{first_user_age} is the Youngest")
    elif second_user_age< first_user_age and second_user_age< third_user_age:
        print(F"{second_user_age} is the Youngest")
    elif third_user_age<second_user_age and third_user_age< first_user_age:
        print(F"{third_user_age} is the Youngest")




Enter The Age of First User 59
Enter The Age of Second User 16
Enter The Age of Third User 19


59 is the Oldest
16 is the Youngest
